sources/lusan/view/common/LogSettings.cpp

```diff
@@ -0,0 +1,136 @@
+﻿#include "lusan/view/common/LogSettings.hpp"
+#include "ui/ui_LogSettings.h"
+#include "lusan/common/LogCollectorClient.hpp"
+#include "lusan/app/LusanApplication.hpp"
+#include "lusan/view/common/WorkspaceManager.hpp"
+#include "lusan/data/log/LogObserver.hpp"
+
+#include <QAbstractItemView>
+#include <QtAssert>
+#include <QAbstractButton>
+#include <QFileDialog>
+#include <QString>
+#include <QMessageBox>
+#include <string>
+
+
+LogSettings::LogSettings(QWidget *parent)
+    : QWidget{parent}
+    , mUi{std::make_unique<Ui::LogSettingsForm>()}
+    , mIpValidator{std::make_unique<QRegularExpressionValidator>(QRegularExpression{R"([0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3})"}, this)}
```

Comment:
I like this idea. It looks very nice. But if the user instead of IP address uses `localhost`, for example? I think, we should not limit with IP address, but should allow names as well.